samigo/samigo-impl/src/java/org/sakaiproject/samigo/impl/SamigoETSProviderImpl.java

```diff
@@ -0,0 +1,441 @@
+import org.jdom.Element;
+import org.jdom.JDOMException;
+import org.jdom.input.SAXBuilder;
+import org.apache.log4j.Logger;
+
+import org.sakaiproject.authz.api.AuthzGroup;
+import org.sakaiproject.authz.api.AuthzGroupService;
+import org.sakaiproject.authz.api.SecurityAdvisor;
+import org.sakaiproject.authz.api.SecurityService;
+import org.sakaiproject.component.api.ServerConfigurationService;
+import org.sakaiproject.emailtemplateservice.model.EmailTemplate;
+import org.sakaiproject.emailtemplateservice.model.RenderedTemplate;
+import org.sakaiproject.emailtemplateservice.service.EmailTemplateService;
+import org.sakaiproject.entity.api.ResourceProperties;
+import org.sakaiproject.event.api.Event;
+import org.sakaiproject.samigo.api.SamigoETSProvider;
+import org.sakaiproject.site.api.Site;
+import org.sakaiproject.site.api.SiteService;
+import org.sakaiproject.tool.api.SessionManager;
+import org.sakaiproject.tool.assessment.facade.PublishedAssessmentFacade;
+import org.sakaiproject.tool.assessment.services.assessment.PublishedAssessmentService;
+import org.sakaiproject.user.api.*;
+import org.sakaiproject.event.api.NotificationService;
+import org.sakaiproject.samigo.util.SamigoConstants;
+import org.sakaiproject.email.api.EmailService;
+import org.sakaiproject.email.api.DigestService;
+
+public class SamigoETSProviderImpl implements SamigoETSProvider {
+    private   static                Logger                      log                             = Logger.getLogger(SamigoETSProviderImpl.class);
+    private                         Map<String,String>          constantValues                  = new HashMap<String, String>();
+    private             final       String                      MULTIPART_BOUNDARY              = "======sakai-multi-part-boundary======";
+    private             final       String                      BOUNDARY_LINE                   = "\n\n--"+MULTIPART_BOUNDARY+"\n";
+    private             final       String                      TERMINATION_LINE                = "\n\n--"+MULTIPART_BOUNDARY+"--\n\n";
+    private             final       String                      MIME_ADVISORY                   = "This message is for MIME-compliant mail readers.";
+    private   static    final       String                      ADMIN                           = "admin";
+    private                         String                      fromAddress                     = "";
+
+    public      void                init                            () {
+        log.info("init()");
+
+        String samigoFromAddress                    = serverConfigurationService.getString("samigo.fromAddress");
+
+        if( samigoFromAddress == null || samigoFromAddress.equals("")){
+            fromAddress                             = serverConfigurationService.getString("setup.request", "no-reply@" + serverConfigurationService.getServerName());
+        } else {
+            fromAddress                             = samigoFromAddress;
+        }
+
```

Comment:
I changed the way the fromAddress was retrieved to allow for defaults to be present.